src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -557,6 +558,25 @@ public Node visitMultiplicativeExpr(JsoniqParser.MultiplicativeExprContext ctx)
         return result;
     }
 
+    @Override
+    public Node visitSimpleMapExpr(JsoniqParser.SimpleMapExprContext ctx) {
+        Expression result = (Expression) this.visitPostFixExpr(ctx.main_expr);
+        if (ctx.map_expr == null || ctx.map_expr.isEmpty()) {
```

Comment:
Noticed this logic for very similar expressions. Will the parser even allow for expression to be empty?

Reply:
Perhaps not... it depends how extra cautious we want to be.